// @filmgardi/videojs-skip-button/src/components/SkipButton.js
import videojs from "video.js";

const Dom = videojs.dom;

class SkipButton {
  constructor(player, options) {
    this.player_ = player;
    this.options = options;
  }
  init() {
    const { player_, options } = this;
    if (options) {
      const { text, from, to, style } = options;
      if (from && to) {
        if (player_) {
          const skipButton = player_.addChild("button");

          skipButton.setAttribute("class", "vjs-fg-skip-button vjs-hidden");
          skipButton.controlText(text);

          this.setupPosition(skipButton.el_);

          if (style) {
            const keys = Object.keys(style);
            if (keys.length > 0) {
              keys.map((key) => {
                skipButton.el().style[key] = style[key];
              });
            }
          }

          const intervalId = skipButton.setInterval(() => {
            if (!player_.paused()) {
              if (player_.currentTime() > from && player_.currentTime() < to) {
                skipButton.show();
              }
              if (player_.currentTime() > to) {
                skipButton.hide();
                clearInterval(intervalId);
              }
            }
          }, 250);
          skipButton.on("click", function () {
            player_.currentTime(to);
          });
        }
      } else {
        videojs.log("Initial data is required.{ from:0s, to:0s }");
      }
    }
  }

  setupPosition(el) {
    // Setup position
    const { offsetH, offsetV } = this.options;

    switch (this.options.position) {
      case "top-left":
        el.style.top = offsetV + "px";
        el.style.left = offsetH + "px";
        break;
      case "top-right":
        el.style.top = offsetV + "px";
        el.style.right = offsetH + "px";
        break;
      case "bottom-left":
        el.style.bottom = offsetV + "px";
        el.style.left = offsetH + "px";
        break;
      case "bottom-right":
        el.style.bottom = offsetV + "px";
        el.style.right = offsetH + "px";
        break;
      default:
        el.style.top = offsetV + "px";
        el.style.right = offsetH + "px";
    }
  }
}
SkipButton.prototype.name = "skipButton";
export default SkipButton;
