// @filmgardi/videojs-skip-button/src/plugin.js
import videojs from "video.js";
// todo:fix this version address
import { version as VERSION } from "../package.json";
import SkipButton from "./components/SkipButton";

const Plugin = videojs.getPlugin("plugin");

// Default options for the plugin.
const defaults = {
  text: "Skip",
  from: 0,
  to: 60,
  position: 'bottom-right',
  offsetH: 46,
  offsetV: 96,
};

/**
 * An advanced Video.js plugin. For more information on the API
 *
 * See: https://blog.videojs.com/feature-spotlight-advanced-plugins/
 */
class Skip extends Plugin {
  /**
   * Create a Skip plugin instance.
   *
   * @param  {Player} player
   *         A Video.js Player instance.
   *
   * @param  {Object} [options]
   *         An optional options object.
   *
   *         While not a core part of the Video.js plugin architecture, a
   *         second argument of options is a convenient way to accept inputs
   *         from your plugin's caller.
   */
  constructor(player, options) {
    // the parent class will add player under this.player
    super(player);

    this.options = videojs.mergeOptions(defaults, options);

    this.player.ready(() => {
      this.player.addClass("vjs-fg-skip");
      this.initial();
      videojs.log("Create Skip-Button");
    });
  }
  /**
   * Initial `skip` logic
   *
   */
  initial() {
    const skipButton = new SkipButton(this.player, this.options);

    skipButton.init();
  }
}

// Define default values for the plugin's `state` object here.
Skip.defaultState = {};

// Include the version number.
Skip.VERSION = VERSION;

// Register the plugin with video.js.
videojs.registerPlugin("skipButton", Skip);

export default Skip;
